tensorflow/compiler/tf2tensorrt/segment/segment.cc

```diff
@@ -681,13 +682,80 @@ void AddSegmentForNode(const grappler::GraphProperties* graph_properties,
 
 }  // namespace
 
+Status ExportNonConversionReportToCSV(
+  string filename,
+  std::map<string, std::map<string, int>>& nonconverted_ops_map) {
+  std::fstream csv_file(filename, std::fstream::out | std::fstream::trunc);
+
+  LOG(WARNING) << "TF-TRT Non-Conversion Report saved at: `" << filename << "`";
+
+  if (!csv_file) {
+      return errors::Internal("Failed to open output file: `", filename, "`");
+  }
+
+  string sep = "|";
+
+  csv_file << "OP Name" << sep << "Reason" << sep << "Count" << std::endl;
+
+  for (auto& op_details : nonconverted_ops_map) {
+      auto op_name = op_details.first;
+      auto op_data = op_details.second;
+
+      for (auto& reject_data : op_data) {
+          auto reason = reject_data.first;
+          auto count = reject_data.second;
+          csv_file << op_name << sep << reason << sep << count << std::endl;
+      }
+  }
+
+  if (!csv_file.bad() && !csv_file.fail()) {
+    return Status::OK();
+  } else {
+    return errors::Internal(
+      "Error closing the file `", filename, "`. "
```

Comment:
The check doesn't close the file, does it? Why the error message says "error closing the file"?

Reply:
Fixed